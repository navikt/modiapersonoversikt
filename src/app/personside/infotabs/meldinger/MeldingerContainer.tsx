import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import { pxToRem } from '../../../../styles/personOversiktTheme';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import { useEffect, useState } from 'react';
import { hasData } from '../../../../rest/utils/restResource';
import { huskForrigeValgtTraad } from '../../../../redux/meldinger/actions';
import { useDispatch } from 'react-redux';
import { useAppState, useRestResource } from '../../../../utils/customHooks';
import { MeldingerDyplenkeRouteComponentProps, useInfotabsDyplenker, useValgtTraadIUrl } from '../dyplenker';
import { withRouter } from 'react-router';
import TraadVisning from './traadvisning/TraadVisning';
import Verktoylinje from './traadvisning/verktoylinje/Verktoylinje';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { ScrollBar, scrollBarContainerStyle } from '../utils/InfoTabsScrollBar';
import { useSokEtterMeldinger } from './utils/meldingerUtils';

interface TraadVisningWrapperProps {
    valgtTraad?: Traad;
    sokeord: string;
    traaderEtterSok: Traad[];
}
const meldingerMediaTreshold = pxToRem(1000);

const MeldingerArticleStyle = styled.article`
    ${scrollBarContainerStyle(meldingerMediaTreshold)};
    @media (min-width: ${meldingerMediaTreshold}) {
        height: 0; /* IE11 */
        flex-grow: 1; /* IE11 */
        display: flex;
        > *:first-child {
            flex: 35% 1 1;
        }
        > *:last-child {
            flex: 65% 1 1;
        }
    }
    position: relative;
`;

function TraadVisningWrapper(props: TraadVisningWrapperProps) {
    if (props.traaderEtterSok.length === 0) {
        return <AlertStripeInfo>Søket ga ingen treff på meldinger</AlertStripeInfo>;
    }
    return (
        <>
            <Verktoylinje valgtTraad={props.valgtTraad} />
            <TraadVisning sokeord={props.sokeord} valgtTraad={props.valgtTraad} />
        </>
    );
}

function MeldingerContainer(props: MeldingerDyplenkeRouteComponentProps) {
    const dispatch = useDispatch();
    const forrigeValgteTraad = useAppState(state => state.meldinger.forrigeValgteTraad);
    const traaderResource = useRestResource(resources => resources.tråderOgMeldinger);
    const dyplenker = useInfotabsDyplenker();
    const traadIUrl = useValgtTraadIUrl(props);
    const [sokeord, setSokeord] = useState('');
    const traaderEtterSok = useSokEtterMeldinger(hasData(traaderResource) ? traaderResource.data : [], sokeord);

    useEffect(
        function visForsteTreffITraadvisning() {
            if (traadIUrl && traaderEtterSok.length > 0 && !traaderEtterSok.includes(traadIUrl)) {
                props.history.push(dyplenker.meldinger.link(traaderEtterSok[0]));
            }
        },
        [traadIUrl, traaderEtterSok, props.history, dyplenker.meldinger]
    );

    useEffect(() => {
        if (!traadIUrl && hasData(traaderResource)) {
            if (traaderResource.data.length === 0) {
                return;
            }
            props.history.push(dyplenker.meldinger.link(forrigeValgteTraad || traaderResource.data[0]));
        } else if (traadIUrl !== forrigeValgteTraad && !!traadIUrl) {
            dispatch(huskForrigeValgtTraad(traadIUrl));
        }
    }, [forrigeValgteTraad, traadIUrl, traaderResource, dispatch, props.history, dyplenker.meldinger]);

    return (
        <RestResourceConsumer<Traad[]>
            getResource={restResources => restResources.tråderOgMeldinger}
            returnOnPending={<CenteredLazySpinner />}
        >
            {data => {
                if (data.length === 0) {
                    return <AlertStripeInfo>Brukeren har ingen meldinger</AlertStripeInfo>;
                }
                return (
                    <MeldingerArticleStyle>
                        <ScrollBar>
                            <TraadListe
                                sokeord={sokeord}
                                setSokeord={setSokeord}
                                traader={data}
                                valgtTraad={traadIUrl}
                            />
                        </ScrollBar>
                        <ScrollBar>
                            <TraadVisningWrapper
                                traaderEtterSok={traaderEtterSok}
                                sokeord={sokeord}
                                valgtTraad={traadIUrl}
                            />
                        </ScrollBar>
                    </MeldingerArticleStyle>
                );
            }}
        </RestResourceConsumer>
    );
}

export default withRouter(MeldingerContainer);
