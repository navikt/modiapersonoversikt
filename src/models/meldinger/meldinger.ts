import { Temagruppe } from '../Temagrupper';
import { JournalforingsSak } from '../../app/personside/infotabs/meldinger/traadvisning/verktoylinje/journalforing/JournalforingPanel';

export interface Traad {
    traadId: string;
    meldinger: Melding[];
}

export interface Melding {
    id: string;
    oppgaveId?: string;
    meldingstype: Meldingstype;
    temagruppe: Temagruppe | null;
    skrevetAvTekst: string;
    journalfortAv?: Saksbehandler;
    journalfortDato?: string;
    journalfortTema?: string;
    journalfortTemanavn?: string;
    journalfortSaksid?: string;
    fritekst: string;
    lestDato?: string;
    status: LestStatus;
    statusTekst?: string;
    opprettetDato: string;
    ferdigstiltDato?: string;
    erFerdigstiltUtenSvar: boolean;
    ferdigstiltUtenSvarAv?: Saksbehandler;
    kontorsperretEnhet?: string;
    kontorsperretAv?: Saksbehandler;
    markertSomFeilsendtAv?: Saksbehandler;
    erDokumentMelding: boolean;
    ferdigstiltUtenSvarDato?: string;
}

export interface Saksbehandler {
    fornavn: string;
    etternavn: string;
    ident?: string;
}

export enum TypeKontakt {
    Telefon = 'telefon',
    Oppmøte = 'oppmote',
    Oppgave = 'oppgave',
    Dokument = 'dokument',
    Monolog = 'monolog',
    MonologUbesvart = 'monlog ubesvart',
    Dialog = 'dialog',
    DialogBesvart = 'dialog besvart'
}

export enum Meldingstype {
    DOKUMENT_VARSEL = 'DOKUMENT_VARSEL',
    OPPGAVE_VARSEL = 'OPPGAVE_VARSEL',
    SPORSMAL_SKRIFTLIG = 'SPORSMAL_SKRIFTLIG',
    SPORSMAL_SKRIFTLIG_DIREKTE = 'SPORSMAL_SKRIFTLIG_DIREKTE',
    SVAR_SKRIFTLIG = 'SVAR_SKRIFTLIG',
    SVAR_OPPMOTE = 'SVAR_OPPMOTE',
    SVAR_TELEFON = 'SVAR_TELEFON',
    DELVIS_SVAR_SKRIFTLIG = 'DELVIS_SVAR_SKRIFTLIG',
    SAMTALEREFERAT_OPPMOTE = 'SAMTALEREFERAT_OPPMOTE',
    SAMTALEREFERAT_TELEFON = 'SAMTALEREFERAT_TELEFON',
    SPORSMAL_MODIA_UTGAAENDE = 'SPORSMAL_MODIA_UTGAAENDE',
    SVAR_SBL_INNGAAENDE = 'SVAR_SBL_INNGAAENDE'
}

export enum LestStatus {
    IkkeLest = 'IKKE_LEST_AV_BRUKER',
    Lest = 'LEST_AV_BRUKER',
    IkkeBesvart = 'IKKE_BESVART'
}

export interface SendReferatRequest {
    fritekst: string;
    temagruppe: Temagruppe;
    meldingstype: Meldingstype.SAMTALEREFERAT_TELEFON | Meldingstype.SAMTALEREFERAT_OPPMOTE;
}

export interface SendSpørsmålRequest {
    fritekst: string;
    sak: JournalforingsSak;
    erOppgaveTilknyttetAnsatt: boolean;
}

export interface ForsettDialogRequest {
    traadId: string;
    behandlingsId: string;
    fritekst: string;
    sak?: JournalforingsSak;
    erOppgaveTilknyttetAnsatt: boolean;
    meldingstype: Meldingstype;
    oppgaveId?: string;
}

export interface SendDelsvarRequest {
    traadId: string;
    behandlingsId: string;
    fritekst: string;
    temagruppe: Temagruppe;
    oppgaveId: string;
}

export interface OpprettHenvendelseRequest {
    traadId: string;
}

export interface OpprettHenvendelseResponse {
    behandlingsId: string;
    oppgaveId?: string;
}

export interface SlaaSammenRequest {
    traader: SlaaSammenTraad[];
    temagruppe: Temagruppe;
}

export interface SlaaSammenTraad {
    oppgaveId: string;
    traadId: string;
}

export interface SlaaSammenResponse {
    nyTraadId: string;
    traader: Traad[];
}
