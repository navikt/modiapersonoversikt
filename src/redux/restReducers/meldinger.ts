import { getMeldinger } from '../../api/meldinger-api';
import { createActionsAndReducer } from './restResource';

const { reducer, action, reload, tilbakestill } = createActionsAndReducer('meldinger');

export function hentMeldinger(fødselsnummer: string) {
    return action(() => getMeldinger(fødselsnummer));
}

export function reloadMeldinger(fødselsnummer: string) {
    return reload(() => getMeldinger(fødselsnummer));
}

export function resetMeldingerResource() {
    return tilbakestill;
}

export default reducer;
