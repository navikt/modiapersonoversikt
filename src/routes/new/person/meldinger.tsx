import { createFileRoute } from '@tanstack/react-router';
import { keepQueryParams } from 'src/utils/keepQueryParams';
import { z } from 'zod';

const meldingerSearchSchema = z.object({
    traadId: z.string().optional()
});

export const meldingerRouteMiddleware = keepQueryParams(['traadId']);

export const Route = createFileRoute('/new/person/meldinger')({
    validateSearch: meldingerSearchSchema,
    search: {
        middlewares: [meldingerRouteMiddleware]
    }
});
