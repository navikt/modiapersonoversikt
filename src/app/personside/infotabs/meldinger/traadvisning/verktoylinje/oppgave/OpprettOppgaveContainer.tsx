import * as React from 'react';
import RestResourceConsumer from '../../../../../../../rest/consumer/RestResourceConsumer';
import { GsakTema } from '../../../../../../../models/meldinger/oppgave';
import OppgaveSkjemaContainer from './OppgaveSkjemaContainer';

interface Props {
    lukkPanel: () => void;
    kontorsperreFunksjon?: () => void;
}

function OpprettOppgaveContainer(props: Props) {
    return (
        <RestResourceConsumer<GsakTema[]> getResource={restResources => restResources.oppgaveGsakTema}>
            {data => (
                <OppgaveSkjemaContainer
                    gsakTema={data}
                    lukkPanel={props.lukkPanel}
                    kontorsperreFunksjon={props.kontorsperreFunksjon}
                />
            )}
        </RestResourceConsumer>
    );
}

export default OpprettOppgaveContainer;
