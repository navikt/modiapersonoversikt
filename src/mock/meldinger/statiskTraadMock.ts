import { LestStatus, Meldingstype, Traad } from '../../models/meldinger/meldinger';
import { saksbehandlerTekst } from '../../app/personside/infotabs/meldinger/utils/meldingerUtils';
import { Temagruppe } from '../../models/Temagrupper';

export const statiskTraadMock: Traad = {
    traadId: 'sg838exr',
    meldinger: [
        {
            id: 'x4k72bwd',
            meldingstype: Meldingstype.SAMTALEREFERAT_OPPMOTE,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'nzxo4q', fornavn: 'Sander', etternavn: 'Hansen' }),
            journalfortAv: { ident: 'vlzaiw', fornavn: 'Adrian', etternavn: 'Pettersen' },
            fritekst:
                'Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.',
            lestDato: '2019-07-12',
            status: LestStatus.Lest,
            opprettetDato: '2019-07-30',
            journalfortDato: '2019-07-11',
            ferdigstiltDato: '2019-07-07',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q1i6ndyj',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xmqgb8', fornavn: 'Frida', etternavn: 'Pedersen' }),
            journalfortAv: { ident: 'ly4pps', fornavn: 'Emil', etternavn: 'Nguyen' },
            fritekst:
                'Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.',
            lestDato: '2019-07-14',
            status: LestStatus.IkkeLest,
            opprettetDato: '2019-08-10',
            journalfortDato: '2019-07-21',
            ferdigstiltDato: '2019-07-16',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q57oayxa',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'e5otkk', fornavn: 'Synne', etternavn: 'Lunde' }),
            journalfortAv: { ident: '3t55yt', fornavn: 'Victoria', etternavn: 'Andresen' },
            fritekst:
                'Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.',
            lestDato: '2019-08-11',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-06',
            journalfortDato: '2019-08-12',
            ferdigstiltDato: '2019-07-12',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: '90l2nkx0',
            meldingstype: Meldingstype.DELVIS_SVAR_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xy70r8', fornavn: 'Kristoffer', etternavn: 'Fredriksen' }),
            journalfortAv: { ident: 'p0jh0d', fornavn: 'Jenny', etternavn: 'Berg' },
            fritekst:
                'Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.',
            lestDato: '2019-07-23',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-13',
            journalfortDato: '2019-07-30',
            ferdigstiltDato: '2019-07-27',
            erFerdigstiltUtenSvar: false,
            erDokumentMelding: false
        }
    ]
};
