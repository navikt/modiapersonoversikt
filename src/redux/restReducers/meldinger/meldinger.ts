import { createRestResourceReducerAndActions } from '../../../rest/utils/restResource';
import { apiBaseUri } from '../../../api/config';
import { AppState } from '../../reducers';
import { Traad } from '../../../models/meldinger/meldinger';

export function getMeldingerFetchUri(state: AppState): string {
    const fnr = state.gjeldendeBruker.f√∏dselsnummer;
    return `${apiBaseUri}/dialog/${fnr}/meldinger`;
}

export default createRestResourceReducerAndActions<Traad[]>('meldinger', getMeldingerFetchUri);
