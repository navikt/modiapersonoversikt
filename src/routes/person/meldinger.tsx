import { createFileRoute } from '@tanstack/react-router';
import MeldingerContainer from 'src/app/personside/infotabs/meldinger/MeldingerContainer';
import { keepQueryParams } from 'src/utils/keepQueryParams';
import { z } from 'zod';

const meldingerSearchSchema = z.object({
    traadId: z.string().optional()
});

export const Route = createFileRoute('/person/meldinger')({
    component: MeldingerContainer,
    validateSearch: meldingerSearchSchema,
    search: {
        middlewares: [keepQueryParams(['traadId'])]
    }
});
