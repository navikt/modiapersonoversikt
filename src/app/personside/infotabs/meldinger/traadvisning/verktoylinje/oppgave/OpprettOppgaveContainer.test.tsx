import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../../../../test/Testprovider';
import OpprettOppgaveContainer from './OpprettOppgaveContainer';
import { statiskTraadMock } from '../../../../../../../mock/meldinger/statiskTraadMock';

test('Viser oppgavecontainer med alt innhold', () => {
    const container = renderer.create(
        <TestProvider>
            <OpprettOppgaveContainer lukkPanel={() => {}} valgtTraad={statiskTraadMock} />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
