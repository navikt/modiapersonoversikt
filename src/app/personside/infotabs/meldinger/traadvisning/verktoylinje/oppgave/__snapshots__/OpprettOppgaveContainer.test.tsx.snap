// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser oppgavecontainer med alt innhold 1`] = `
Array [
  .c5 {
  position: relative;
  border: none;
  padding: 0;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c5:after {
  border-bottom: 1px #b7b1a9 solid;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
}

.c5:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c5:hover {
  opacity: 0.8;
}

.c5:hover:after {
  content: '';
  border-color: #0067c5;
}

.c1 {
  position: relative;
}

.c3 {
  position: absolute !important;
  bottom: 0.4rem;
  right: 0.4rem;
}

.c2 {
  position: relative;
  margin-bottom: 1rem;
}

.c2 .skjemaelement {
  margin-bottom: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 button {
  margin: 0;
}

.c0 .inputPanelGruppe__inner {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .inputPanelGruppe__inner > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<form
    onSubmit={[Function]}
  >
    <div
      className="c0"
    >
      <div
        className="skjemaelement"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Tema
        </label>
        <div
          className="selectContainer input--fullbredde"
        >
          <select
            className="skjemaelement__input"
            disabled={false}
            id="Helt tilfeldig ID"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Velg tema
            </option>
            <option
              value="AGR"
            >
              Ajourhold - Grunnopplysninger
            </option>
            <option
              value="AAP"
            >
              Arbeidsavklaringspenger
            </option>
          </select>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
      <div
        className="skjemaelement"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Gjelder
        </label>
        <div
          className="selectContainer input--fullbredde"
        >
          <select
            className="skjemaelement__input"
            disabled={false}
            id="Helt tilfeldig ID"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Ingen tema valgt
            </option>
          </select>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
      <div
        className="skjemaelement"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Type oppgave
        </label>
        <div
          className="selectContainer input--fullbredde"
        >
          <select
            className="skjemaelement__input"
            disabled={false}
            id="Helt tilfeldig ID"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Ingen tema valgt
            </option>
          </select>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        className="c1"
        role="combobox"
      >
        <div
          className="c2"
        >
          <div
            className="skjemaelement"
          >
            <label
              className="skjemaelement__label"
              htmlFor="downshift-0-input"
            >
              Velg enhet
            </label>
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-label="Laster data"
              aria-labelledby="downshift-0-label"
              autoComplete="off"
              className="skjemaelement__input input--fullbredde"
              id="downshift-0-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
          <svg
            aria-label="Laster innhold"
            className="spinner c3 spinner--s"
            focusable="false"
            height={32}
            kind="spinner"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 50 50"
            width={32}
          >
            <title>
              Venter...
            </title>
            <circle
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke="#bdbab7"
              strokeWidth="5"
              xmlns="http://www.w3.org/2000/svg"
            />
            <circle
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke="#7f756c"
              strokeDasharray="50 155"
              strokeLinecap="round"
              strokeWidth="5"
            />
          </svg>
        </div>
      </div>
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        className="c1"
        role="combobox"
      >
        <div
          className="c2"
        >
          <div
            className="skjemaelement"
          >
            <label
              className="skjemaelement__label"
              htmlFor="downshift-1-input"
            >
              Velg ansatt
            </label>
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-label="Laster data"
              aria-labelledby="downshift-1-label"
              autoComplete="off"
              className="skjemaelement__input input--fullbredde"
              id="downshift-1-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
          <svg
            aria-label="Laster innhold"
            className="spinner c3 spinner--s"
            focusable="false"
            height={32}
            kind="spinner"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 50 50"
            width={32}
          >
            <title>
              Venter...
            </title>
            <circle
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke="#bdbab7"
              strokeWidth="5"
              xmlns="http://www.w3.org/2000/svg"
            />
            <circle
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke="#7f756c"
              strokeDasharray="50 155"
              strokeLinecap="round"
              strokeWidth="5"
            />
          </svg>
        </div>
      </div>
      <div
        className="inputPanelGruppe"
      >
        <fieldset
          className="skjema__fieldset"
        >
          <legend
            className="skjema__legend"
          >
            Prioritet
          </legend>
          <div
            className="inputPanelGruppe__inner"
          >
            <label
              className="inputPanel radioPanel"
              htmlFor="Helt tilfeldig ID"
            >
              <input
                aria-checked={false}
                checked={false}
                className="inputPanel__field"
                id="Helt tilfeldig ID"
                name="Prioritet"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="HOY"
              />
              <span
                className="inputPanel__label"
              >
                HÃ¸y
              </span>
            </label>
            <label
              className="inputPanel radioPanel inputPanel--checked"
              htmlFor="Helt tilfeldig ID"
            >
              <input
                aria-checked={true}
                checked={true}
                className="inputPanel__field"
                id="Helt tilfeldig ID"
                name="Prioritet"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="NORM"
              />
              <span
                className="inputPanel__label"
              >
                Normal
              </span>
            </label>
            <label
              className="inputPanel radioPanel"
              htmlFor="Helt tilfeldig ID"
            >
              <input
                aria-checked={false}
                checked={false}
                className="inputPanel__field"
                id="Helt tilfeldig ID"
                name="Prioritet"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="LAV"
              />
              <span
                className="inputPanel__label"
              >
                Lav
              </span>
            </label>
          </div>
        </fieldset>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
      <div
        className="skjemaelement textarea__container"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Beskrivelse
        </label>
        <div
          className="textarea--medMeta__wrapper"
        >
          <textarea
            className="skjemaelement__input textarea--medMeta"
            id="Helt tilfeldig ID"
            onChange={[Function]}
            style={
              Object {
                "height": "30px",
              }
            }
            type="text"
            value=""
          />
          <p
            className="textarea--medMeta__teller"
          >
            <span
              aria-live="polite"
              className="teller-tekst"
            >
              Du har 2000 tegn igjen
            </span>
          </p>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
        <div
          aria-hidden="true"
          className="textareamirror"
        />
      </div>
      <div
        className="c4"
      >
        <button
          className="knapp knapp--hoved"
          disabled={false}
          type="submit"
        >
          Send
        </button>
        <button
          className="c5"
          onClick={[Function]}
          type="button"
        >
          Avbryt
        </button>
      </div>
    </div>
  </form>,
  .c0 {
  padding-top: .8rem;
  color: #d0021b;
}

<div
    aria-live="polite"
    className="c0"
  />,
]
`;
